Ask user for tab to start

1-alex98 opened the pull request:
User is asked what tab he wants to start in instead of taking last tab as starting point.
Fixes #1723

![start_tab](https://user-images.githubusercontent.com/19996713/82908280-0eaf3500-9f68-11ea-9bbb-1252d691ef87.gif)

1-alex98 committed:
Ask user for tab to start

User is asked what tab he wants to start in instead of taking last tab as starting point.
Fixes #1723

BlackYps commented:
The new comboboxes should have the same style as the other comboboxes in the client. e.g. the Database box right below in the settings

1-alex98 commented:
already changed that, just the gif is old

BlackYps commented:
A test is failing:
```
com.faforever.client.main.StartTabChooseControllerTest > unnecessary Mockito stubbings FAILED
    org.mockito.exceptions.misusing.UnnecessaryStubbingException
```

1-alex98 commented:
yes gotta fix that

anihilnine commented:
this is great. it would be best if we didnt launch it until the newsfeed is up

BlackYps commented on src/main/resources/i18n/messages.properties:
```diff
@@ -523,7 +523,7 @@ settings.giveFeedback=Open
 settings.general=General
 settings.language=Language
-settings.general.rememberLastTab=Remember my last opened tab
+settings.general.startTab=Select the tab to starts at
```
Typo.
should be "Select the tab to start at"

BlackYps commented on src/main/resources/i18n/messages.properties:
```diff
@@ -759,3 +759,9 @@ replay.ignoreMapNotFound=Ignore and start replay
 replay.abortAfterMapNotFound=Abort and show more info
 replay.mapDownloadFailed=Map download failed for this replay
 replay.mapDownloadFailed.wannaContinue=We could not find the map mentioned in the replay file. However the game might still know where to find it. Consider this as a warning, that this replay might be broken.
+startTab.selectTab=Select the tab you want to starts at:
```
See above

Brutus5000 commented on src/main/java/com/faforever/client/main/event/NavigateEvent.java:
```diff
@@ -1,7 +1,9 @@
+import lombok.EqualsAndHashCode;
 import lombok.Getter;
 
+@EqualsAndHashCode
 public class NavigateEvent {
   @Getter
   private final NavigationItem item;
```
I'd suggest using `@Value` and `@AllArgsConstructor`.
Then remove `@EqualsAndHashCode`, remove `@Getter private final` and remove the constructor

Brutus5000 commented on src/main/java/com/faforever/client/main/event/NavigationItem.java:
```diff
@@ -1,16 +1,20 @@
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
 import java.util.HashMap;
 
+@RequiredArgsConstructor
 public enum NavigationItem {
```
This class also looks like a candidate for `@Value` (see above)

1-alex98 replied:
Not possible on enums

Brutus5000 commented on src/main/java/com/faforever/client/notification/ImmediateNotification.java:
```diff
@@ -1,33 +1,34 @@
  * action or dismisses it. The notification consists of a title, a text, an optional image and zero or more actions.
  */
+@RequiredArgsConstructor
 public class ImmediateNotification {
```
This class also looks like a candidate for @Value (see above)

1-alex98 committed:
Ask user for tab to start

User is asked what tab he wants to start in instead of taking last tab as starting point.
Fixes #1723
